### PR TITLE
Add Tags on Request form

### LINKED ISSUE #20: Implement add Tags on request
**User story**
As a user, I want to be able to add request tag based on my input so that I can present my request more clearly and volunteers easily filter with added tag

**Acceptance criteria**
- User can add suggested tags by clicking tags (Done)
- Add tag input area. so the user can insert new request tags on request form
   1. when a user clicked `add a new tag` , then show `input` area, 
   2. when a user clicked `add` button then new selected tag is shown to form. 
- Ensure the result items on business have new tags

### PR DESCRIPTION
**Pull request by anhyesun**
#### [Add Tags on Request form](#)

**Checklist**
- [x] Run `npm run lint:fix` or `yarn lint:fix`

**What and how**
* Request form has its own tags that provide sample tags list
* When a user clicked tags and add the submit button, the request adds tags and business updates its tags.

**Screenshots**
addTags

### CODE REVIEW

**Comment by anhyesun** [comment]
Linter changes the spacing in Homepage.js file. Plz ignore or delete changes in Homepage.js file

File: client/src/components/Form/RequestForm.js [hunk 1]
```diff
@@ -56,7 +63,16 @@ const Form = ({
...
-                <Tags canAdd />
+                <div className={classes.tags}>
+                    {tags.map(({ label, selected }, i) => (
+                        <div key={i}>
+                            {selected
+                                ? <SelectedChip label={label} onClick={() => selectTag(i)} />
+                                : <UnselectedChip label={label} onClick={() => selectTag(i)} />}
+                        </div>
+                    ))}
+                    { <DottedChip /> }
```
Comment (yehee) [inline client/src/components/Form/RequestForm.js]: Any reason for extra bracket?

File: client/src/components/Form/RequestForm.js [hunk 2]
```diff
@@ -56,7 +63,16 @@ const Form = ({
                     ))}
                 </Select>
                 <Text>Request Details</Text>
-                <Tags canAdd />
+                <div className={classes.tags}>
+                    {tags.map(({ label, selected }, i) => (
+                        <div key={i}>
```
Comment (yehee) [inline client/src/components/Form/RequestForm.js]: do we need this wrapper?

File: client/src/components/Form/RequestForm.js [hunk 3]
```diff
@@ -5,36 +5,43 @@ import {
...
         setDetails('')
-        setTags([])
+        setTags(defaultTags)
         closeForm('request')
     }
 
     const onSubmit = e => {
-        addRequest({ business, details, tags })
+        addRequest({ business, details, tags: tags.filter(({ selected }) => selected).map(({ label }) => ({ label })) })
+        addBusiness({ tags: tags.filter(({ selected }) => selected).map(({ label }) => ({ label })) })
```
Comment (yehee) [inline client/src/components/Form/RequestForm.js]: what does this do? should call `updateBusiness` after retrieving `_id` of the business this request should belong to

File: client/src/components/Form/RequestForm.js (hunk 1 above)
Comment (anhyesun) [reply client/src/components/Form/RequestForm.js]: Done!

File: client/src/components/Form/RequestForm.js (hunk 2 above)
Comment (anhyesun) [reply client/src/components/Form/RequestForm.js]: Changed!

File: client/src/components/Form/RequestForm.js (hunk 3 above)
Comment (anhyesun) [reply client/src/components/Form/RequestForm.js]: Took off that line. Seems like updateBusiness needs to be implemented on actions as well. I'll update it when Edit business task is done :)

Comment (yehee) [reply client/src/components/Form/RequestForm.js]: that works! I'd just make sure to note it down somewhere

**Review by yehee — APPROVED** [review]

**Pull request merged**